qiskit/visualization/text.py

```diff
@@ -776,19 +776,35 @@ def wire_names(self, with_initial_state=False):
         qubit_labels = []
         if self.layout is None:
             for bit in self.qubits:
-                label = "{name}_{index}: " + initial_qubit_value
+                # check the size of register
+                if len(self.qubits) > 1:
+                    label = "{name}_{index}: " + initial_qubit_value
+                    index = self.bit_locations[bit]["index"]
+                else:
+                    label = "{name}: " + initial_qubit_value
+
                 if self.bit_locations[bit]["register"] is not None:
-                    qubit_labels.append(
-                        label.format(
-                            name=self.bit_locations[bit]["register"].name,
-                            index=self.bit_locations[bit]["index"],
-                            physical="",
+                    # different format for non-singleton register
+                    if len(self.qubits) > 1:
+                        qubit_labels.append(
+                            label.format(
+                                name=self.bit_locations[bit]["register"].name,
+                                index=index,
+                                physical="",
+                            )
+                        )
+                    else:
+                        qubit_labels.append(
+                            label.format(
+                                name=self.bit_locations[bit]["register"].name,
+                                physical="",
+                            )
```

Comment:
Do you think you it is possible to do something similar to what has been mentioned in comment at line 819?